src/test/scala/org/ergoplatform/network/DecodingUtils.scala

```diff
@@ -0,0 +1,61 @@
+package org.ergoplatform.network
+
+import java.nio.ByteBuffer
+
+// Common functions used in parsers, collected in one place to be easily understood and rewritten into other languages
+trait DecodingUtils {
+
+  /**
+    * Read unsigned byte
+    */
+  def getUByte(buf: ByteBuffer): Int = getByte(buf) & 0xFF
+
+  /**
+    * Read signed byte
+    */
+  def getByte(buf: ByteBuffer): Byte = buf.get
+
+  /**
+    * Reading up to 64 bits VLQ-encoded value. The value is considered to be signed.
+    * Can be read less than 64 bytes!
+    * See https://en.wikipedia.org/wiki/Variable-length_quantity
+    */
+  def getULong(buf: ByteBuffer): Long = {
```

Comment:
Why not call `VLQReader.getULong()`? Otherwise, it adds an overhead to keep these two implementations in sync.
This question is also relevant to other methods in this trait.

Reply:
The tests are done in particular as examples for external application developers. Thus it was made intentionally to collect all the needed functions in one file to avoid need for traversing different classes.